src/components/PhoneField/PhoneField.tsx

```diff
@@ -0,0 +1,338 @@
+import React, {useState, useEffect, useCallback} from 'react';
+
+import {TextField} from '../TextField';
+import {Button} from '../Button';
+import {Popover} from '../Popover';
+import {ActionList} from '../ActionList';
+
+import styles from './PhoneField.scss';
+
+export interface Country {
+  /** The country flag */
+  image: string;
+  /** Country name */
+  countryName: string;
+  /** Country area code */
+  countryCode: string;
+  countryAlphaCode?: string;
+  /** Phone number display format */
+  displayFormat: number[];
+  /** Possible area codes for a country. Used to distinguish between countries with same country codes */
+  areaCodes?: number[];
+  population?: number;
+  formatter?(): void;
+}
+
+export interface PhoneFieldProps {
+  /** Placeholder text  */
+  placeholder?: string;
+  /** Textfield label  */
+  labelName?: string;
+  /** Hides the label for accessibility */
+  labelHidden?: boolean;
+  /** Is textfield optional */
+  optional?: boolean;
+  /** Error message */
+  errorMessage?: string;
+  /** Country list in dropdown */
+  countries: Country[];
+  /** The search bar in the dropdown is present */
+  searchBar?: boolean;
```

Comment:
Do we have a use case for showing/hiding the search bar? If possible I'd like to avoid adding options with not existing use case other than "it seemed handy"